clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -169,8 +169,14 @@ public void run(long now) {
         RecordAccumulator.ReadyCheckResult result = this.accumulator.ready(cluster, now);
 
         // if there are any partitions whose leaders are not known yet, force metadata update
-        if (result.unknownLeadersExist)
+        if (!result.unknownLeaderTopics.isEmpty()) {
+            // Unknown topics are removed from metadata to avoid unnecessary metadata fetch for
```

Comment:
This comment is a little confusing to me. Are you saying that the unknownLeaderTopics are the same topics that were removed from the cache explicitly because of UNKNOWN_TOPIC_OR_PARTITION errors? Are both errors for the topic possible at the same time?